### PR TITLE
Stable timestamps

### PR DESCRIPTION
**Pull request by tsibley**
One doc change for `set_properties` and a behaviour change when packaging up the zip file to set zip file member timestamps.

The commit messages explain the rationale.

### CODE REVIEW

File: lib/Excel/Writer/XLSX.pm [hunk 1]
```diff
@@ -448,15 +448,19 @@ The properties that can be set are:
     title
     subject
     author
+    last_author
     manager
     company
     category
```
Comment (jmcnamara) [inline lib/Excel/Writer/XLSX.pm]: W

Comment (tsibley) [inline lib/Excel/Writer/XLSX.pm]: > W

I'm assuming you meant to type more?  :-)

Comment (jmcnamara) [inline lib/Excel/Writer/XLSX.pm]: The W was an attempt to either zoom or scroll on my phone. I'll delete the comment(s).

**Comment by jmcnamara** [comment]
> The commit messages explain the rationale.

The CONTRIBUTING.md explicitly says:

> ...
> All patches and pull requests are welcome but must start with an issue tracker.

> **Getting Started**

>Pull requests and new feature proposals must start with an issue tracker. This serves as the focal point for the design discussion.
>...

This looks like a bug report so it should have started with a bug report and then moved to a PR.

Anyway, I'll make some comments anyway but I'm starting off in the wrong mood.

File: lib/Excel/Writer/XLSX/Workbook.pm [hunk 2]
```diff
@@ -1092,10 +1093,17 @@ sub _store_workbook {
...
     for my $filename ( @xlsx_files ) {
         my $short_name = $filename;
         $short_name =~ s{^\Q$tempdir\E/?}{};
-        $zip->addFile( $filename, $short_name );
+        my $member = $zip->addFile( $filename, $short_name );
+
+        # Set zip member timestamp to match the workbook's timestamp, so that
+        # generated files are stable when the workbook's "created" property is
+        # manually set.
+        $member->setLastModFileDateTimeFromUnix( $createtime );
```
Comment (jmcnamara) [inline lib/Excel/Writer/XLSX/Workbook.pm]: Overall the commit looks good and the commit message is good, which puts me in a bit less of a bad mood. However, it still isn't clear to me why this change is either required (for a bug) or desired (for a feature request). Could you clarify?

Also, what does stable mean here?

File: lib/Excel/Writer/XLSX/Workbook.pm [hunk 3]
```diff
@@ -1092,10 +1093,17 @@ sub _store_workbook {
     );
 
     # Store the xlsx component files with the temp dir name removed.
+    my $createtime = timegm(@{ $self->{_doc_properties}{created} || $self->{_createtime} });
```
Comment (jmcnamara) [inline lib/Excel/Writer/XLSX/Workbook.pm]: I would prefer this on 2 lines. Something like:

```perl
my @created_time = @{ $self->{_doc_properties}->{created} || $self->{_createtime} };

my $created_time = timegm(@created_time);
```

File: lib/Excel/Writer/XLSX.pm [hunk 4]
```diff
@@ -448,15 +448,19 @@ The properties that can be set are:
     title
```
Comment (jmcnamara) [inline lib/Excel/Writer/XLSX.pm]: > I was disappointed to not see "created" in the list initially and went
about looking for other solutions before finally digging into the source
and realizing the it was actually supported!

So this is a feature request then?

File: lib/Excel/Writer/XLSX.pm [hunk 5]
```diff
@@ -448,15 +448,19 @@ The properties that can be set are:
...
     category
     keywords
     comments
+    created
     status
     hyperlink_base
 
 See also the C<properties.pl> program in the examples directory of the distro.
+Note that the C<created> property must be an array reference which matches the
+return values of L<perlfunc/gmtime> in list context, e.g. C<< created => [ gmtime() ] >>.
```
Comment (jmcnamara) [inline lib/Excel/Writer/XLSX.pm]: The  `last_author` and `created` properties are deliberately undocumented since no-one has ever asked for them. If someone had asked I would have made them public.

What is your use cases for `last_author` and `created`?

**Comment by tsibley** [comment]
The first commit could be considered a documentation bug report.  The second commit is somewhere between bug report and feature request.

If you want me to make issues of them retroactively, that's fine.

**Comment by jmcnamara** [comment]
> If you want me to make issues of them retroactively, that's fine.

Please do.

File: lib/Excel/Writer/XLSX/Workbook.pm (hunk 3 above)
Comment (tsibley) [inline lib/Excel/Writer/XLSX/Workbook.pm]: I will fix up the commit and repush with that change.

File: lib/Excel/Writer/XLSX/Workbook.pm (hunk 2 above)
Comment (tsibley) [inline lib/Excel/Writer/XLSX/Workbook.pm]: Stable here means "the same when generated with the same inputs".

I'll elaborate on the reason for this in the issue I create.

**Comment by tsibley** [comment]
I've created two issues for this PR, #167 and #168, and further explained things there.
